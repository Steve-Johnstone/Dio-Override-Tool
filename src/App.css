body {
	font-family: Open Sans, sans-serif !important;
	margin: 25px auto 25px auto;
}
.override-list {
	height: 600px;
	width: 80%;
	display: flex;
	flex-direction: column;
	flex-wrap: wrap;
}

.override-list li {
	margin: 0em 1.5em 0.6em 0em;
	cursor: pointer;
}

.override-list li:hover {
	background-color: rgb(241, 236, 236);
}

.header {
	color: white;
	background-color: #d32f2f;
	font-size: 1.8em;
	min-height: 90px;
	text-align: left;
}

#search-input-field {
	color: white;
	background-color: rgba(255, 255, 255, 0.2);
	border-radius: 5px;
	width: 100%;
	padding: 10px;
}

#search-input-field::placeholder {
	color: white;
}

.search-bar {
	width: 100%;
	min-width: 250px;
}

.search-icon {
	position: absolute;
	margin-top: 3px;
	min-width: 40px;
	margin-left: 210px;
}

#search-bar-container {
	margin-right: 100px;
}

.footer-section {
	background-color: #e7e7e7;
	padding: 15px;
}

.footer-url-display {
	background-color: rgba(255, 255, 255, 0.75);
	padding: 3px;
	margin: 0px 30px 10px 50px;
}

.override-selected {
	background-color: #fefdca;
}

.footer-override-display {
	margin: 10px 30px 10px 23px;
}

.footer-override-display li {
	background-color: rgba(255, 255, 255, 0.75);
	display: inline-block;
	margin-right: 10px;
	margin-bottom: 10px;
	padding: 5px;
}

.footer-override-list li {
	cursor: pointer;
}

.main-page-container {
	width: 60%;
	border: 2px solid black;
	margin-left: auto;
	margin-right: auto;
}

.override-display-conatiner {
	width: 60%;
	border: 2px solid black;
	margin-left: auto;
	margin-right: auto;
}
