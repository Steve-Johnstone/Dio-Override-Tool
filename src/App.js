import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        Override Tool
      </header>
    </div>
  );
}

export default App;
