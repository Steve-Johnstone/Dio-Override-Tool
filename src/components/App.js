import React, { useState } from 'react';
import MainPage from './MainPage.js';
import Overrides from './Overrides.js';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import '../App.css';

function App() {
	const [url, setUrl] = useState('');
	const [selectedOverrides, setSelectedOverrides] = useState([]);

	return (
		<Router>
			<Routes>
				<Route
					exact
					path='/'
					element={
						<MainPage
							url={url}
							setUrl={setUrl}
							setSelectedOverrides={setSelectedOverrides}
						/>
					}
				/>
				<Route
					exact
					path='/overrides'
					element={<Overrides selectedOverrides={selectedOverrides} />}
				/>
			</Routes>
		</Router>
	);
}

export default App;
