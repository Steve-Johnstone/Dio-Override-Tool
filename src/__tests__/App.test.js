import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../components/App';

describe('App.js', () => {
	it('should display the DOM correctly', () => {
		const tree = renderer.create(<App />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render the main screen correctly', () => {
		render(<App />);

		//Check for Title
		expect(screen.getByText('Dio Override Tool')).toBeInTheDocument();

		//Check for input field
		expect(screen.getByText('Local URL of the page')).toBeInTheDocument();

		//Check for GO button
		expect(screen.getByRole('button', { name: /Go/i })).toBeInTheDocument();
	});
});
