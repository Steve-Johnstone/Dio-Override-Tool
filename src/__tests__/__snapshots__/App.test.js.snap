// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js should display the DOM correctly 1`] = `
<div
  className="main-page-container"
>
  <div
    className="header level-item"
  >
    <div>
      <p
        className="ml-6"
      >
        Dio Override Tool
      </p>
    </div>
  </div>
  <section>
    <div
      className="container p-4"
    >
      <div>
        <form
          className="field py-2"
        >
          <div
            className="control"
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <label
              className="label"
              htmlFor="text"
            >
              Local URL of the page
            </label>
            <div>
              <input
                className="input is-link level-item"
                name="url"
                onChange={[Function]}
                placeholder="www.url.com"
                type="text"
              />
            </div>
            <button
              className="button is-info mt-3"
              name="Go"
              onClick={[Function]}
            >
              <a
                href="/overrides"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                GO
              </a>
            </button>
          </div>
        </form>
      </div>
    </div>
  </section>
</div>
`;
